src/main/java/com/sequenceiq/cloudbreak/domain/Stack.java

```diff
@@ -63,39 +77,67 @@
                         + "WHERE (s.account= :account AND s.publicInAccount= true) "
                         + "OR s.owner= :user"),
         @NamedQuery(
                 name = "Stack.findAllInAccount",
                 query = "SELECT s FROM Stack s "
+                        + "LEFT JOIN FETCH s.resources "
+                        + "LEFT JOIN FETCH s.instanceMetaData "
+                        + "LEFT JOIN FETCH s.instanceGroups "
                         + "WHERE s.account= :account "),
         @NamedQuery(
                 name = "Stack.findByAmbari",
                 query = "SELECT s from Stack s "
+                        + "LEFT JOIN FETCH s.resources "
+                        + "LEFT JOIN FETCH s.instanceMetaData "
+                        + "LEFT JOIN FETCH s.instanceGroups "
                         + "WHERE s.ambariIp= :ambariIp"),
         @NamedQuery(
                 name = "Stack.findOneByName",
                 query = "SELECT c FROM Stack c "
+                        + "LEFT JOIN FETCH c.resources "
+                        + "LEFT JOIN FETCH c.instanceMetaData "
+                        + "LEFT JOIN FETCH c.instanceGroups "
                         + "WHERE c.name= :name and c.account= :account"),
         @NamedQuery(
                 name = "Stack.findByIdInAccount",
                 query = "SELECT s FROM Stack s "
+                        + "LEFT JOIN FETCH s.resources "
+                        + "LEFT JOIN FETCH s.instanceMetaData "
+                        + "LEFT JOIN FETCH s.instanceGroups "
                         + "WHERE s.id= :id and ((s.account= :account and s.publicInAccount=true) or s.owner= :owner)"),
         @NamedQuery(
                 name = "Stack.findByNameInAccount",
                 query = "SELECT s FROM Stack s "
+                        + "LEFT JOIN FETCH s.resources "
+                        + "LEFT JOIN FETCH s.instanceMetaData "
+                        + "LEFT JOIN FETCH s.instanceGroups "
                         + "WHERE s.name= :name and ((s.account= :account and s.publicInAccount=true) or s.owner= :owner)"),
         @NamedQuery(
                 name = "Stack.findByNameInUser",
                 query = "SELECT t FROM Stack t "
+                        + "LEFT JOIN FETCH t.resources "
+                        + "LEFT JOIN FETCH t.instanceMetaData "
+                        + "LEFT JOIN FETCH t.instanceGroups "
                         + "WHERE t.owner= :owner and t.name= :name"),
         @NamedQuery(
                 name = "Stack.findByCredential",
-                query = "SELECT t FROM Stack t WHERE t.credential.id= :credentialId")
+                query = "SELECT t FROM Stack t "
+                        + "LEFT JOIN FETCH t.resources "
+                        + "LEFT JOIN FETCH t.instanceGroups "
+                        + "LEFT JOIN FETCH t.instanceMetaData "
+                        + "WHERE t.credential.id= :credentialId ")
```

Comment:
I know these left joins solve the LazyInitializationExceptions but this is the same like eager initialization that causes a lot of unnecessary fetches from the db. I'm not sure if it can cause some performance problems when fetching thousand node clusters, @keyki please check it out.